### PR TITLE
treewide: do not define/capture unused variables

### PR DESCRIPTION
**Pull request by tchaikov**
these warnings are found by Clang-17 after removing `-Wno-unused-lambda-capture` and `-Wno-unused-variable` from the list of disabled warnings in `configure.py`.

### CODE REVIEW

**Comment by tchaikov** [comment]
i am happy to chop this change into smaller pieces if this helps with review.

**Review by avikivity — APPROVED** [review]

**Review by nyh — CHANGES_REQUESTED** [review]
Looks good, though I can't say with 100% certainty I didn't miss everything. I think I found one typo.
By the way, now that you don't need these options in config.py, don't you want to also remove them?

File: api/compaction_manager.cc [hunk 1]
```diff
@@ -65,8 +65,8 @@ void set_compaction_manager(http_context& ctx, routes& r) {
     });
 
     cm::get_pending_tasks_by_table.set(r, [&ctx] (std::unique_ptr<request> req) {
-        return ctx.db.map_reduce0([&ctx](replica::database& db) {
-            return do_with(std::unordered_map<std::pair<sstring, sstring>, uint64_t, utils::tuple_hash>(), [&ctx, &db](std::unordered_map<std::pair<sstring, sstring>, uint64_t, utils::tuple_hash>& tasks) {
+        return ctx.db.map_reduce0([&](replica::database& db) {
```
[the remark this answers is not in the document]
Comment (tchaikov) [reply api/compaction_manager.cc]: it is indeed a typo.  fixed and repushed.

**Comment by tchaikov** [comment]
changelog :

-  s/&//: it was a typo.

**Comment by tchaikov** [comment]
> Looks good, though I can't say with 100% certainty I didn't miss everything. I think I found one typo.
> By the way, now that you don't need these options in config.py, don't you want to also remove them?

good point. i do want to remove them. as they help to improve the code to help us to identify potential bugs. will drop them in another commit in this changeset.

**Review by nyh — APPROVED** [review]

**Comment by tchaikov** [comment]
i will remove `-Wno-unused-variable` and `-Wno-unused-lambda-capture` in a follow-up change. also, i noticed there are a handful warnings not fixed by this changeset. will fix them in anther PR.

**Review by denesb — APPROVED** [review]

**Comment by denesb** [comment]
This PR was queued as 0cb842797a104e922fcb81a27ad749d560d9af75 but it doesn't have the `Closes` reference.

**Comment by tchaikov** [comment]
@denesb ack! thanks for the reminder. i will rebase this changeset once the test completes.

**Comment by denesb** [comment]
The "Closes" tag should have been added automatically by our merge script. Not sure what went wrong.

**Comment by avikivity** [comment]
Committed as 0cb842797a104e922fcb81a27ad749d560d9af75

**Pull request closed without merging**